bubble_sort.rb

```diff
@@ -0,0 +1,17 @@
+# bubble_sort
+
+def bubble_sort(arr)
+  arr.length.times do
+    i = 0
+    while i < arr.length - 1
+      if arr[i] > arr [i + 1]
+        (arr[i], arr[i + 1] = arr[i + 1], arr[i])
+      else arr = arr
+      end
+      i += 1
+    end
+  end
+  prints arr
+end
+
```

Comment:
- [ ] In methods or functions that involve all logic, you do not print/puts the result, instead, you return the result.
In the highlighted code above, you need to return the output and remember that Ruby implicitly returns the last variable in a method. So you do not need to add the return statement.
**This should be done for both methods.**

Note: If you wish to see the result of your output in the console, you can add the puts statement when you call the method. Like so -
`puts bubble_sort([2, 7, 5, 1, 10, 5, 39, 20, 6, 8, 17])`
```
sorted = bubble_sort_by([2, 7, 5, 6, 1]) do |left, right|
  left - right
end
puts sorted
```